src/RotationMatrix.java

```diff
@@ -0,0 +1,184 @@
+     * @param angle Angle of counterclockwise rotation in degrees.
+     * @param axis  Vector3D that represents the axis of rotation.
+     */
+    public RotationMatrix(double angle, Vector3D axis) {
+        double c = Math.cos(Math.toRadians(angle));
+        double s = Math.sin(Math.toRadians(angle));
+
+        if (axis.norm() != 1) {
+            axis = axis.normalize();
+        }
+
+        matrix[0][0] = c + axis.x() * axis.x() * (1 - c);
+        matrix[0][1] = axis.x() * axis.y() * (1 - c) - axis.z() * s;
+        matrix[0][2] = axis.x() * axis.z() * (1 - c) + axis.y() * s;
+
+        matrix[1][0] = axis.y() * axis.x() * (1 - c) + axis.z() * s;
+        matrix[1][1] = c + axis.y() * axis.y() * (1 - c);
+        matrix[1][2] = axis.y() * axis.z() * (1 - c) - axis.x() * s;
+
+        matrix[2][0] = axis.z() * axis.x() * (1 - c) - axis.y() * s;
+        matrix[2][1] = axis.z() * axis.y() * (1 - c) + axis.x() * s;
+        matrix[2][2] = c + axis.z() * axis.z() * (1 - c);
+    }
+
+    /**
+     * Constructor with array.
+     *
+     * @param matrix A double[3][3] that contains the elements of the matrix.
+     */
+    public RotationMatrix(double[][] matrix) {
+
+        if (matrix == null) {
+            throw new NullPointerException();
+        }
+
+
+        for (int i = 0; i < SIZE; i++) {
+            for (int j = 0; j < SIZE; j++) {
+                this.matrix[i][j] = matrix[i][j];
+            }
+        }
+    }
+
+    /**
+     * Constructor for basic rotation around X axis.
+     *
+     * @param angle Angle in degrees of counterclockwise rotation around X axis.
+     */
+    public static RotationMatrix basicRotationX(double angle) {
+        return new RotationMatrix(angle, new Vector3D(1, 0, 0));
+    }
+
+    /**
+     * Constructor for basic rotation around Y axis.
+     *
+     * @param angle Angle in degrees of counterclockwise rotation around Y axis.
+     */
+    public static RotationMatrix basicRotationY(double angle) {
+        return new RotationMatrix(angle, new Vector3D(0, 1, 0));
+    }
+
+    /**
+     * Constructor for basic rotation around Z axis.
+     *
+     * @param angle Angle in degrees of counterclockwise rotation around Z axis.
+     */
+    public static RotationMatrix basicRotationZ(double angle) {
+        return new RotationMatrix(angle, new Vector3D(0, 0, 1));
+    }
+
+
+    /**
+     * Method to rotate a vector.
+     *
+     * @param vector Vector3D to rotate.
+     * @return Result of the rotation.
+     */
+    public Vector3D rotate(Vector3D vector) {
+        double[] result = new double[]{0, 0, 0};
+
+        for (int i = 0; i < SIZE; i++) {
+            for (int j = 0; j < SIZE; j++) {
+                result[i] += this.getElement(i, j) * vector.getElement(j);
+            }
+        }
+
+        return new Vector3D(result);
+    }
+
+    /**
+     * Method to assert if to Rotation Matrices are equal.
+     *
+     * @return Returns True if they are equal, and False otherwise.
```

Comment:
AFAIK you use uppercase to describe the classes and lowercase to describe the instances (hence the comment would be "Returns true if they are equal, and false otherwise"). But I'm not sure what is the standard for Java comments.